### ISSUE TITLE
[Enhance Search filter] <Hello i have implemented enhance search filters for our input searh box>

### ISSUE THREAD

**Issue by GeekkyCoder** [Vijaykv5/SnapGrid#122]
I want to enhance the input search box 

i have implmenteted and enhanced the search filters for our input box, i just need your permisson so i can push changes

1) Users can choose from our given list of keywords to find images, the search filter also suggest the possible keywords and also be able to show no options, if no keyword that user has typed matched to our list of keywords. 

2) Room for improvement for asynchronous actions, can also get our list of keywords from our database and if they are predefined by us and have them on db , we can dynamically populate our keywords inside  search filter as the user start typing in the search box.

Output:
![Screenshot (463)](https://github.com/Vijaykv5/SnapGrid/assets/104452510/581286f2-b663-43a1-b00f-68ebf05491d6)
![Screenshot (464)](https://github.com/Vijaykv5/SnapGrid/assets/104452510/fefad10f-0681-4b1c-8be3-53e21474f3a3)
![Screenshot (465)](https://github.com/Vijaykv5/SnapGrid/assets/104452510/4c481a9d-7b78-4bc6-9afe-ddc46c6e2dd6)

please allow me this task

**Comment by Vijaykv5** [comment]
Yeah that's a good thing 

but does the suggestions come for every item searched?

also i would like to improve the UI of suggestion a little bit more

**Comment by GeekkyCoder** [comment]
yes the suggestion popups for user, and also user can type inside this input and can also choose to open the dropdown and see possible keywords

**Comment by Vijaykv5** [comment]
Ok cool. Go for it 🚀
